### PR TITLE
chore: audit and test for nils

### PR DESCRIPTION
**Pull request by pgautier404**
This commit adds validation to prevent nil values in request structs and adds testing to verify the added validation across request types.

Addresses #192 and #178

### CODE REVIEW

File: momento/requester.go [hunk 1]
```diff
@@ -70,29 +70,61 @@ type hasTTL interface {
...
+	return nameBytes, nil
+}
+
 func prepareCacheName(r hasCacheName) (string, error) {
 	return prepareName(r.cacheName(), "Cache name")
 }
 
 func prepareKey(r hasKey) ([]byte, error) {
-	key := r.key().asBytes()
+	err := buildError(momentoerrors.InvalidArgumentError, "key cannot be nil or empty", nil)
```
Comment (eaddingtonwhite) [inline momento/requester.go:109]: nit: but seems slightly off to build up error we might return at top like this especially in high traffic place like `prepareKey` maybe better just create inline for each time returned on line 112 and 117 so doesnt get allocated every time this function called? Similar to what your doing in `prepareField`.

File: momento/requester.go [hunk 2]
```diff
@@ -101,8 +133,15 @@ func prepareField(r hasField) ([]byte, error) {
 }
 
 func prepareFields(r hasFields) ([][]byte, error) {
+	err := buildError(momentoerrors.InvalidArgumentError, "fields cannot be nil or empty", nil)
```
Comment (eaddingtonwhite) [inline momento/requester.go:136]: same thought here on error being inline on 138 and 143

File: momento/requester.go [hunk 3]
```diff
@@ -155,24 +190,22 @@ func prepareTTL(r hasTTL, defaultTtl time.Duration) (uint64, error) {
...
-			nil,
+		return 0, buildError(
+			momentoerrors.InvalidArgumentError, "ttl must be a non-zero positive value", nil,
 		)
 	}
 	return uint64(ttl.Milliseconds()), nil
 }
 
 func momentoValuesToPrimitiveByteList(i []Value) ([][]byte, momentoerrors.MomentoSvcErr) {
+	err := buildError(momentoerrors.InvalidArgumentError, "values may not be nil", nil)
```
Comment (eaddingtonwhite) [inline momento/requester.go:201]: also here

File: momento/requester.go (hunk 1 above)
Comment (pgautier404) [reply momento/requester.go:109]: That's fine. I figured it was DRYer to do it this way and relatively low impact, but I am fine with the tiny but of duplication.

Comment (pgautier404) [reply momento/requester.go]: Fixed in 0c32ba1

**Review by eaddingtonwhite — APPROVED** [review]
Nice work! Tests Looking great! 🧼

**Pull request merged**